snapcraft/internal/lxd.py

```diff
@@ -76,6 +77,11 @@ def __init__(self, *, output, source, project_options,
                 'Unrecognized server architecture {}'.format(kernel))
         self._host_arch = deb_arch
         self._image = 'ubuntu:xenial/{}'.format(deb_arch)
+        # Use a temporary folder the 'lxd' snap can access
+        lxd_common = os.path.expanduser(
+            os.path.join('~', 'snap', 'lxd', 'common'))
+        os.makedirs(lxd_common, exist_ok=True)
+        self.tmp = tempfile.mkdtemp(prefix='snapcraft', dir=lxd_common)
```

Comment:
please rename `tmp` to `tmp_dir` for clarity.
While we are at it, maybe even rename `lxd_common` to `lxd_common_dir`